### PR TITLE
Add model generation with a DNN (python only)

### PR DESCRIPTION
**Pull request by cassand**

### CODE REVIEW

File: WORKSPACE [hunk 1]
```diff
@@ -220,6 +221,14 @@ pypi_repository(
     version = "3.4.0",
 )
 
+pypi_wheel_repository(
```
Comment (FunkySayu) [inline WORKSPACE:224]: I am not sure to understand why you moved from the tensorflow defined in third_party to the tensorflow defined in pypi. This might cause some versionning issues if at some point, we try to re-use the C++ part of tensorflow.

What's the reason of this change?

File: WORKSPACE [hunk 2]
```diff
@@ -220,6 +221,14 @@ pypi_repository(
...
+    name = "pydef_tensorflow",
```
Comment (FunkySayu) [inline WORKSPACE]: `name = "pydep_tensorflow",`

File: powerspikegg/computation_models/match/graph.py [hunk 3]
```diff
@@ -25,69 +25,78 @@ class GraphBuilder:
     def __init__(self, input_size):
         self.input_size = input_size
 
-    def add_hidden_layer(self, data, input_size, output_size, name,
-                         keep_probability, is_training):
+    def add_hidden_layer(self, data, units, name, is_training):
+        """ Create a simple layer for a neural network
```
Comment (FunkySayu) [inline powerspikegg/computation_models/match/graph.py:29]: Note: thank you for the comments, that really helps.

File: powerspikegg/computation_models/match/graph.py [hunk 4]
```diff
@@ -25,69 +25,78 @@ class GraphBuilder:
...
-    def inference(self, data):
+    def createNetwork(self, data, layers, is_training):
+    	""" Create a deep neural network
+
+            Args:
+                data: A tensor with the input data
+                layers: An array with the size of each layer 
+                is_training: A tensor containing a boolean to indicate if the model
+                             is training
+	    """
```
Comment (FunkySayu) [inline powerspikegg/computation_models/match/graph.py]: re-indent end of comments

File: powerspikegg/computation_models/match/graph.py [hunk 5]
```diff
@@ -25,69 +25,78 @@ class GraphBuilder:
...
         Args:
-            logits: Logits tensor, float - [batch_size, 1].
+            logits: Logits tensor, float - [batch_size].
             labels: Labels tensor, float - [batch_size].
 
         Returns:
             loss: Loss tensor of type float.
         """
-        return tf.reduce_sum(tf.pow(logits - labels, 2), name='square_error')
+        loss = tf.losses.mean_squared_error(labels, logits)
```
Comment (FunkySayu) [inline powerspikegg/computation_models/match/graph.py]: one-line this: `return tf.losses.mean_squared_error(labels, logits)`

File: powerspikegg/computation_models/match/graph.py [hunk 6]
```diff
@@ -25,69 +25,78 @@ class GraphBuilder:
...
-                                           keep_probability, is_training)
-            last_size = layer_size
-            current = current + 1
+	        norm = tf.layers.batch_normalization(data, training=is_training)
+	        hidden = tf.layers.dense(inputs=norm, units=layer_size, activation=tf.nn.relu)
+        
         return hidden
 
-    def inference(self, data):
+    def createNetwork(self, data, layers, is_training):
```
Comment (FunkySayu) [inline powerspikegg/computation_models/match/graph.py]: `create_network` https://google.github.io/styleguide/pyguide.html?showone=Naming#Naming

**Review by FunkySayu — APPROVED** [review]
some nits to fix before merging

File: WORKSPACE (hunk 1 above)
Comment (cassand) [reply WORKSPACE:224]: It greatly simplify the workflow as we can build everything with Bazel, the time to compile the computation module is also reduced from 30 min to a minute. (+ should work on mac)

As we don't have a working version for the c++ serving yet we have no advantage in recompiling Tensorflow.

For now it might be better to focus on creating and training the model and only later on the performance of the serving.

Also note:
* The pip module can use the gpu
* Tensorflow might have a better integration with Bazel soon (no configure or git subrepository needed)

Comment (FunkySayu) [reply WORKSPACE:224]: Ack, thank your for the details.

@ArchangelX360 you might be able to compile the project :p

Comment (FunkySayu) [reply WORKSPACE:224]: I just thought, you should delete the `/third_party/tensorflow` then. And ensure everything is working fine.

Comment (cassand) [reply WORKSPACE:224]: Sure, I will make the test, however it might be useful to keep the subrepository in case we want to work on the c++ module later (only for development purposes) we could also remove the submodule initialization from the configure script (to be considered). 

Otherwise we can move the whole c++ module to an other branch and clean the related part from the develop branch (submodule, c++ target)

Comment (cassand) [reply WORKSPACE:224]: Everything is working without the Tensorflow subrepository (I might need the pyserving part for further tests)
